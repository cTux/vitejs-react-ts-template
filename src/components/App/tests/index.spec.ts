import { AppDriver } from './driver';

describe('App', () => {
  let driver: AppDriver;

  beforeEach(() => {
    driver = new AppDriver();
  });

  it('should match snapshot', async () => {
    await driver.render();
    expect(driver.component.innerHTML).toMatchSnapshot();
  });

  it('should change state and re-render', async () => {
    await driver.render();
    expect(driver.counter.component.textContent).toEqual('Counter: 0');
    driver.button.trigger.click();
    expect(driver.counter.component.textContent).toEqual('Counter: 1');
  });
});
