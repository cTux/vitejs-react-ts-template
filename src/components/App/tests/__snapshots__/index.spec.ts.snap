// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should match snapshot 1`] = `"<h3 class=\\"_counter_30b361\\">Counter: 0</h3><button id=\\"clickMe\\">Click me!</button>"`;
